test_scripts/AppIconsFolder/ATF_N_AppIconsAmountToRemove_larger_than_icons_amount.lua

```diff
@@ -0,0 +1,326 @@
+  local i =1
+  while sizeToFull > 326360 do
+    os.execute("sleep " .. tonumber(10))
+    local copyFileToAppIconsFolder = assert( os.execute( "cp files/icon.png " .. tostring(config.pathToSDL) .. tostring(AppIconsFolder) .. "/icon" .. tostring(i) ..".png"))
+    i = i + 1
+    if copyFileToAppIconsFolder ~= true then
+      commonFunctions:userPrint(31, " Files are not copied to " .. tostring(AppIconsFolder))
+    end
+    sizeAppIconsFolderInBytes = dSize(config.pathToSDL .. tostring(AppIconsFolder))
+    sizeToFull = 1048576 - sizeAppIconsFolderInBytes
+    if i > 50 then
+      commonFunctions:userPrint(31, " Loop is breaking due to a lot of iterations ")
+      break
+    end
+  end
+end
+
+local function checkFunction()
+  local applicationFileExistsResult = checkFileExistence(applicationFileToCheck)
+  local firstFileExistsResult = checkFileExistence(firstOldFileToCheck)
+  local secondFileExistsResult = checkFileExistence(secondOldFileToCheck)
+  local thirdFileExistsResult = checkFileExistence(thirdOldFileToCheck)
+  local aHandle = assert( io.popen( "ls " .. config.pathToSDL .. "Icons/" , 'r'))
+  local listOfFilesInStorageFolder = aHandle:read( '*a' )
+  commonFunctions:userPrint(32, "Content of storage folder: " ..tostring("\n" ..listOfFilesInStorageFolder))
+
+  if applicationFileExistsResult ~= true then
+    commonFunctions:userPrint(31, tostring(RAIParameters.appID) .. " icon is absent")
+    status = false
+  end
+  if 
+    firstFileExistsResult ~= false or
+    secondFileExistsResult ~= false or
+    thirdFileExistsResult ~= false then
+    commonFunctions:userPrint(31,"Oldest icons are not deleted from AppIconsFolder")
+    status = false
+  end
+    return status
+end
+
+--[[ Preconditions ]]
+commonSteps:DeleteLogsFileAndPolicyTable()
+commonFunctions:newTestCasesGroup("Preconditions")
+
+function Test.Precondition_stopSDL()
+  StopSDL()
+end 
+
+function Test.Precondition_configureAppIconsFolder()
+  checkSDLPathValue()
+  local appIconsFolderValueToReplace = "Icons"
+  local stringToReplace = "AppIconsFolder = " .. tostring(appIconsFolderValueToReplace) .. "\n"
+  file = assert(io.open(SDLini, "r"))
+  if file then
+    fileContent = file:read("*all")
+    local matchResult = string.match(fileContent, "AppIconsFolder%s-=%s-.-%s-\n")
+    if matchResult ~= nil then
+      fileContentUpdated  =  string.gsub(fileContent, matchResult, stringToReplace)
+      file = assert(io.open(SDLini, "w"))
+      file:write(fileContentUpdated)
+    else
+      commonFunctions:userPrint(31, "'AppIconsFolder = value' is not found. Expected string finding and replacing value with " .. tostring(appIconsFolderValueToReplace))
+    end
+    file:close()
+  end
+end
+ 
+function Test.Precondition_configureAppIconsFolderMaxSize()
+  checkSDLPathValue()
+  local appIconsFolderMaxSizeValueToReplace = "1048576"
+  local stringToReplace = "AppIconsFolderMaxSize = " .. tostring(appIconsFolderMaxSizeValueToReplace) .. "\n"
```

Comment:
@aarotar 
I propose to use one of existing functions from already connected modules for edit values in SDL ini file `commonFunctions:SetValuesInIniFile()`